### PR TITLE
DomainTools New Playbooks

### PR DESCRIPTION
**Pull request by RamboV**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by moin-loginsoft** [comment]
During packaging, we encountered some issues and are currently investigating the exact problem. Please refer to the below screenshot for details. Any suggestions or insights would be greatly appreciated.

image

**Comment by RamboV** [comment]
During packaging we encountered some issues and are currently investigating the exact problem. Please refer to the below screenshot for details. Any suggestions or insights would be greatly appreciated.

image

**Comment by v-prasadboke** [comment]
Hello @RamboV, Please create a custom table named as DomainToolsDomainEnrichment_CL at location 
https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables

**Comment by RamboV** [comment]
Hello @v-prasadboke, Added custom table DomainToolsDomainEnrichment_CL.json at specified location.

**Comment by v-prasadboke** [comment]
> Hello @RamboV, Please create a custom table named as DomainToolsDomainEnrichment_CL at location https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables
> 
> Custom connector located in Playbook folder, looks like Function app. If it is a Function app please move it to Data Connector folder and add Data connector file as well.

Please have a look at the below mentioned point

**Comment by RamboV** [comment]
> > Hello @RamboV, Please create a custom table named as DomainToolsDomainEnrichment_CL at location https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables
> > Custom connector located in Playbook folder, looks like Function app. If it is a Function app please move it to Data Connector folder and add Data connector file as well.
> 
> Please have a look at the below mentioned point

Hello @v-prasadboke , it is a FuntionApp custom connector, very similar to the below
https://github.com/Azure/Azure-Sentinel/tree/3a43f4e66e6d156bc7dbc4ebc420108050fcca6b/Solutions/AWS_IAM/Playbooks/AWS_IAM_FunctionAppConnector
https://github.com/Azure/Azure-Sentinel/tree/3a43f4e66e6d156bc7dbc4ebc420108050fcca6b/Solutions/AWSAthena/Playbooks/CustomConnector/AWSAthena_FunctionAppConnector

Please let us know if you still need it to be moved

**Comment by v-prasadboke** [comment]
Will check on it @RamboV.

File: Solutions/DomainTools/Playbooks/CustomConnector/DomainTools_FunctionAppConnector/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+                    ],
+                    "properties": {
+                        "FUNCTIONS_EXTENSION_VERSION": "~4",
+                        "FUNCTIONS_WORKER_RUNTIME": "python",
+                        "APPINSIGHTS_INSTRUMENTATIONKEY": "[reference(resourceId('Microsoft.insights/components', variables('FunctionName')), '2015-05-01').InstrumentationKey]",
+                        "APPLICATIONINSIGHTS_CONNECTION_STRING": "[reference(resourceId('microsoft.insights/components', variables('FunctionName')), '2015-05-01').ConnectionString]",
+                        "AzureWebJobsStorage": "[concat('DefaultEndpointsProtocol=https;AccountName=', toLower(variables('FunctionName')),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', toLower(variables('FunctionName'))), '2019-06-01').keys[0].value, ';EndpointSuffix=',toLower(variables('StorageSuffix')))]",
+                        "APIKey": "[parameters('DomainToolsAPIKey')]",
+                        "APIUsername": "[parameters('DomainToolsAPIUsername')]",
+                        "WEBSITE_RUN_FROM_PACKAGE": "https://github.com/moin-loginsoft/domaintoolsaction/blob/main/DomainToolsFunctionApp.zip?raw=true"
```
Comment (v-prasadboke) [inline Solutions/DomainTools/Playbooks/CustomConnector/DomainTools_FunctionAppConnector/azuredeploy.json:154]: Hello @RamboV 
The hyperlink which is pointing to a zip is stored somewhere else. 

Should this point to the zip located in custom connector folder or it is correct

**Comment by v-prasadboke** [comment]
Hello @RamboV, The PR is still failing for a KQL validation. Can you check it once.

**Comment by RamboV** [comment]
Hello @v-prasadboke , we digged the issue, will submit the changes tomorrow, your change now doesn't solve the pipeline issues.

**Comment by v-prasadboke** [comment]
> Hello @v-prasadboke , we digged the issue, will submit the changes tomorrow, your change now doesn't solve the pipeline issues.

Got it @RamboV.

**Comment by RamboV** [comment]
> > Hello @v-prasadboke , we digged the issue, will submit the changes tomorrow, your change now doesn't solve the pipeline issues.
> 
> Got it @RamboV.

Hello @v-prasadboke , we have updated the parser, all checks have passed, kindly check

**Comment by v-prasadboke** [comment]
Hello @RamboV, All looks fine to me. I have one final requirement. 
The logo which is being used in createui and input file is in png format. 

Can you please provide us a logo in svg  format and which should be less than or =  5kb

**Comment by RamboV** [comment]
> Hello @RamboV, All looks fine to me. I have one final requirement. The logo which is being used in createui and input file is in png format.
> 
> Can you please provide us a logo in svg format and it should be less than or = 5kb

@v-prasadboke is it ok it is like 6kb, do you want us to send here?

**Comment by RamboV** [comment]
> > Hello @RamboV, All looks fine to me. I have one final requirement. The logo which is being used in createui and input file is in png format.
> > Can you please provide us a logo in svg format and it should be less than or = 5kb
> 
> @v-prasadboke is it ok it is like 6kb, do you want us to send here?

DomainTools Logo Color

**Comment by v-prasadboke** [comment]
You can add it in the Logos folder

**Comment by RamboV** [comment]
> You can add it in the Logos folder

Hello @v-prasadboke done ,

**Review by v-prasadboke — APPROVED** [review]

**Review by v-sabiraj — APPROVED** [review]

**Pull request merged**